### PR TITLE
Update TaskRunStatus.ResourcesResult to be more generic.

### PR DESCRIPTION
**Pull request by dlorenc**
# Changes

This commit extends ResourcesResult to work for more than just ImageOutputs,
but moving to a key/value map instead of a fixed key of "digest".

This also updates the image_digest_exporter to write data in both formats
for backwards compatibility, and updates our e2e test to use this mechanism.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```
- The Name and Digest fields on TaskRunStatus.ResourcesResult are deprecated and are replaced by the new Key and ResourceRef fields.
- Both sets of fields are present in this release, but a future release will remove the legacy fields.
```

### CODE REVIEW

**Comment by dlorenc** [comment]
Ref: https://github.com/tektoncd/pipeline/issues/1392

File: test/kaniko_task_test.go [hunk 1]
```diff
@@ -34,32 +31,41 @@ import (
...
 	taskSpecOps := []tb.TaskSpecOp{
 		tb.TaskInputs(tb.InputsResource("gitsource", v1alpha1.PipelineResourceTypeGit)),
+		tb.TaskOutputs(tb.OutputsResource("builtImage", v1alpha1.PipelineResourceTypeImage)),
 	}
 	stepOps := []tb.StepOp{
 		tb.StepArgs(
 			"--dockerfile=/workspace/gitsource/integration/dockerfiles/Dockerfile_test_label",
 			fmt.Sprintf("--destination=%s", repo),
 			"--context=/workspace/gitsource",
+			"--oci-layout-path=/builder/home/image-outputs/builtImage",
```
Comment (bobcatfish) [inline test/kaniko_task_test.go]: 🎉

File: test/kaniko_task_test.go [hunk 2]
```diff
@@ -152,40 +160,6 @@ func TestKanikoTaskRun(t *testing.T) {
...
-	for _, initContainer := range p.Spec.InitContainers {
-		containers = append(containers, initContainer.Name)
-	}
-	for _, container := range p.Spec.Containers {
-		containers = append(containers, container.Name)
-	}
-
-	return getContainerLogs(c, pod, namespace, containers...)
-}
-
```
Comment (bobcatfish) [inline test/kaniko_task_test.go]: 😍

File: test/pipelinerun_test.go [hunk 3]
```diff
@@ -70,7 +70,7 @@ func TestPipelineRun(t *testing.T) {
 
 			for _, res := range getFanInFanOutGitResources(namespace) {
 				if _, err := c.PipelineResourceClient.Create(res); err != nil {
-					t.Fatalf("Failed to create Pipeline Resource `%s`: %s", kanikoResourceName, err)
+					t.Fatalf("Failed to create Pipeline Resource `%s`: %s", kanikoGitResourceName, err)
```
Comment (bobcatfish) [inline test/pipelinerun_test.go]: this seems like an odd change - i wonder if the wrong string was being used here previously?

File: pkg/apis/pipeline/v1alpha1/resource_types.go [hunk 4]
```diff
@@ -164,8 +164,13 @@ type PipelineResourceBinding struct {
 
 // PipelineResourceResult used to export the image name and digest as json
 type PipelineResourceResult struct {
+	// Name and Digest are deprecated.
 	Name   string `json:"name"`
 	Digest string `json:"digest"`
+	// These will replace Name and Digest soon
```
Comment (bobcatfish) [inline pkg/apis/pipeline/v1alpha1/resource_types.go]: maybe create an issue and add the # here?

**Comment by dlorenc** [comment]
OK, docs are updated and ready to go.

**Comment by dlorenc** [comment]
Rebased!

**Pull request merged**